05/app.js

```diff
@@ -0,0 +1,64 @@
+function Student( firstName , secondName ) {
+
+    this.firstName = firstName,
+    this.secondName = secondName,
+
+    this.grades = {};
+}
+
+
+Student.prototype.addGrade = function(subject, grades) {
+
+
+      if(typeof this.grades[subject] === 'undefined') {
+          this.grades[subject] = [];
+          this.grades[subject].push(grades);
+
+      } else {
+          this.grades[subject].push(grades);
+      }
+}
+
+Student.prototype.getAverageGrade = function(subject) {
+
+    if( !( typeof this.grades[subject] === 'undefined' ) ) {
+
+        let sum = 0;
+        for(let i=0; i <= this.grades[subject].length -1; i++) {
+
+            sum = sum + this.grades[subject][i]
+        }
```

Comment:
Dla czytelności przed `for` można by utworzyć zmienną `const subjectGrades = this.grades[subject]`.